kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/KroxyliciousST.java

```diff
@@ -94,18 +99,22 @@ void moreThanOneServicesInDifferentNamespaces(String namespace) {
         bootstrap = kroxylicious2.getBootstrap(clusterName);
         assertThat(bootstrap).withFailMessage("bootstrap " + bootstrap + " does not contain the corresponding namespace " + newNamespace)
                 .contains(newNamespace);
-        produceAndConsumeMessage(newNamespace, bootstrap, randomTopicName());
+        produceAndConsumeMessage(newNamespace, bootstrap, randomTopicName(), CompressionType.NONE);
     }
 
     private void produceAndConsumeMessage(String namespace, String bootstrap) {
-        produceAndConsumeMessage(namespace, bootstrap, topicName);
+        produceAndConsumeMessage(namespace, bootstrap, topicName, CompressionType.NONE);
+    }
+
+    private void produceAndConsumeMessage(String namespace, String bootstrap, CompressionType compressionType) {
+        produceAndConsumeMessage(namespace, bootstrap, topicName, compressionType);
     }
 
-    private void produceAndConsumeMessage(String namespace, String bootstrap, String topicName) {
+    private void produceAndConsumeMessage(String namespace, String bootstrap, String topicName, CompressionType compressionType) {
         int numberOfMessages = 1;
 
         LOGGER.atInfo().setMessage("And a kafka Topic named {}").addArgument(topicName).log();
-        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 1);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 1, compressionType);
 
         LOGGER.atInfo().setMessage("When {} messages '{}' are sent to the topic '{}'").addArgument(numberOfMessages).addArgument(MESSAGE).addArgument(topicName).log();
         KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
```

Comment:
I think we should be passing compressionType through here as well to test producer based compression.